src/gauging.jl

```diff
@@ -59,172 +86,96 @@ function vidal_gauge(
-function vidal_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  verbose=false,
-  svd_kwargs...,
-)
-  ψψ = norm_network(ψ)
-  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+  cache![] = update(cache![]; cache_update_kwargs...)
 
-  mts = belief_propagation(pψψ; niters, target_precision=target_canonicalness, verbose)
-  return vidal_gauge(
-    ψ, pψψ, mts; eigen_message_tensor_cutoff, regularization, svd_kwargs...
-  )
+  return update(ψv, cache![]; kwargs...)
 end
 
 """Transform from an ITensor in the Vidal Gauge (bond tensors) to the Symmetric Gauge (partitionedgraph, message tensors)"""
-function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
-  ψsymm = copy(ψ)
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-  ψsymm_mts = default_bp_cache(pψψsymm)
+function symmetric_gauge(ψv::VidalITensorNetwork)
+  ψsymm = copy(itensornetwork(ψv))
+
+  bp_cache = default_norm_cache(ψsymm)
+  mts = messages(bp_cache)
 
   for e in edges(ψsymm)
     vsrc, vdst = src(e), dst(e)
-    pe = partitionedge(pψψsymm, NamedEdge((vsrc, 1) => (vdst, 1)))
-    root_S = sqrt_diag(bond_tensors[e])
+    pe = partitionedge(bp_cache, (vsrc, 1) => (vdst, 1))
+    root_S = sqrt_diag(bond_tensor(ψv, e))
     setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vsrc]), vsrc)
     setindex_preserve_graph!(ψsymm, noprime(root_S * ψsymm[vdst]), vdst)
 
-    ψsymm_mts[pe], ψsymm_mts[reverse(pe)] = copy(ITensor[dense(bond_tensors[e])]),
-    copy(ITensor[dense(bond_tensors[e])])
+    set!(mts, pe, copy(ITensor[dense(bond_tensor(ψv, e))]))
+    set!(mts, reverse(pe), copy(ITensor[dense(bond_tensor(ψv, e))]))
   end
 
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
+  bp_cache = default_norm_cache(ψsymm)
 
-  return ψsymm, pψψsymm, ψsymm_mts
+  return ψsymm, set_messages(bp_cache, mts)
 end
 
 """Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices from the Vidal Gauge)"""
-function symmetric_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  svd_kwargs...,
-)
-  ψ_vidal, bond_tensors = vidal_gauge(
-    ψ;
-    eigen_message_tensor_cutoff,
-    regularization,
-    niters,
-    target_canonicalness,
-    svd_kwargs...,
-  )
-
-  return vidal_to_symmetric_gauge(ψ_vidal, bond_tensors)
-end
+function symmetric_gauge(ψ::ITensorNetwork; kwargs...)
+  ψv = VidalITensorNetwork(ψ)
+  ψv = update(ψv; kwargs...)
```

Comment:
I have now written it so that this works. Although I still have the constructor `VidalITensorNetwork(ψ, bp_cache::BeliefPropagationCache)` in the code which this function passes to once it has done the relevant cache updating